fault-tolerance/fault-tolerance/src/main/java/io/helidon/faulttolerance/MetricsUtils.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2025 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.faulttolerance;
+
+import java.util.List;
+import java.util.function.Supplier;
+
+import io.helidon.common.LazyValue;
+import io.helidon.common.config.Config;
+import io.helidon.metrics.api.Counter;
+import io.helidon.metrics.api.Gauge;
+import io.helidon.metrics.api.MeterRegistry;
+import io.helidon.metrics.api.Metrics;
+import io.helidon.metrics.api.MetricsFactory;
+import io.helidon.metrics.api.Tag;
+import io.helidon.metrics.api.Timer;
+
+import static io.helidon.faulttolerance.FaultTolerance.FT_METRICS_ENABLED;
+import static io.helidon.metrics.api.Meter.Scope.VENDOR;
+
+@SuppressWarnings("unchecked")
+class MetricsUtils {
+
+    private static final LazyValue<MetricsFactory> METRICS_FACTORY = LazyValue.create(MetricsFactory::getInstance);
+    private static final LazyValue<MeterRegistry> METRICS_REGISTRY = LazyValue.create(Metrics::globalRegistry);
+
+    private static volatile Boolean enableMetrics;
+
+    private MetricsUtils() {
+    }
+
+    /**
+     * Looks for the metrics enabled flag in config and caches result. FT metrics
+     * are disabled by default.
+     *
+     * @return value of metrics flag
+     */
+    static boolean enableMetrics() {
+        if (enableMetrics == null) {
+            Config config = FaultTolerance.config();
+            enableMetrics = config.get(FT_METRICS_ENABLED).asBoolean().orElse(false);
+        }
+        return enableMetrics;
+    }
+
+    static <T extends Number> void gaugeBuilder(String name, Supplier<T> supplier, Tag... tags) {
+        Gauge.Builder<T> builder = METRICS_FACTORY.get().gaugeBuilder(name, supplier).scope(VENDOR);
```

Comment:
Not essential to change this, but for future reference you can avoid the explicit reference to the metrics factory by using, for example, `Gauge.builder(...)` or `Counter.builder(...)`, etc.